src/vcpkg/commands.build.cpp

```diff
@@ -311,10 +311,10 @@ namespace vcpkg
     }
 
 #if defined(_WIN32)
-    static ZStringView to_vcvarsall_target(StringView cmake_system_name)
+    static ZStringView to_vcvarsall_target(StringView cmake_system_name, StringView cmake_system_version)
     {
-        if (cmake_system_name.empty()) return "";
-        if (cmake_system_name == "Windows") return "";
+        if (cmake_system_name.empty()) return {cmake_system_version.data(), cmake_system_version.size()};
+        if (cmake_system_name == "Windows") return {cmake_system_version.data(), cmake_system_version.size()};
         if (cmake_system_name == "WindowsStore") return "store";
```

Comment:
I have not analyzed everything this code is doing, but returning "store" in one path and returning a version number in another path is almost certainly wrong.

Moreover, this does not meet the ZStringView requirements because it's returning a pointer that it claims is null terminated with inputs that are not null terminated.